### PR TITLE
Switch header convention from `#include "SDL.h"` to `#include <SDL3/SDL.h>`

### PR DESCRIPTION
**Pull request by slouken**
I ran this script in the include directory:
```sh
sed -i '' -e 's,#include "\(SDL.*\)",#include <SDL3/\1>,' *.h
```

I ran this script in the src directory:
```sh
for i in ../include/SDL3/SDL*.h
do hdr=$(basename $i)
   if [ x"$(echo $hdr | egrep 'SDL_main|SDL_name|SDL_test|SDL_syswm|SDL_opengl|SDL_egl|SDL_vulkan')" != x ]; then
        find . -type f -exec sed -i '' -e 's,#include "\('$hdr'\)",#include <SDL3/\1>,' {} \;
    else
        find . -type f -exec sed -i '' -e '/#include "'$hdr'"/d' {} \;
    fi
done
```

Fixes https://github.com/libsdl-org/SDL/issues/6575

### CODE REVIEW

**Comment by slouken** [comment]
@madebr, why would this change result in Win32 tests failing to link?
```
[  1%] Linking C executable checkkeys.exe
/opt/local/bin/cmake -E cmake_link_script CMakeFiles/checkkeys.dir/link.txt --verbose=1
/opt/local/bin/cmake -E rm -f CMakeFiles/checkkeys.dir/objects.a
/opt/local/bin/i686-w64-mingw32-ar qc CMakeFiles/checkkeys.dir/objects.a @CMakeFiles/checkkeys.dir/objects1.rsp
/opt/local/bin/i686-w64-mingw32-gcc -Wl,--whole-archive CMakeFiles/checkkeys.dir/objects.a -Wl,--no-whole-archive -o checkkeys.exe -Wl,--out-implib,libcheckkeys.dll.a -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles/checkkeys.dir/linklibs.rsp
/opt/local/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: cannot find -lSDL3main: No such file or directory
collect2: error: ld returned 1 exit status
```

File: sdl3-config.in [hunk 1]
```diff
@@ -46,7 +46,7 @@ while test $# -gt 0; do
       echo @SDL_VERSION@
       ;;
     --cflags)
-      echo -I@includedir@/SDL3 @SDL_CFLAGS@
+      echo -I@includedir@ -I@includedir@/SDL3 @SDL_CFLAGS@
```
Comment (madebr) [inline sdl3-config.in:49]: Is it okay if I remove all `-I@includedir@/SDL3` from all sdl3 config scripts (pkg-config, sdl3-config, cmake)?

Comment (madebr) [reply sdl3-config.in:49]:
```suggestion
      echo -I@includedir@ @SDL_CFLAGS@
```

Comment (sezero) [reply sdl3-config.in:49]: > Is it okay if I remove all `-I@includedir@/SDL3` from all sdl3 config scripts (pkg-config, sdl3-config, cmake)? Or is it too harsh?

I don't think that it'd be harsh.

Comment (slouken) [reply sdl3-config.in:49]: Yes, that's fine, although I'm about to remove sdl3-config, so you can skip that one.

**Pull request merged**